### PR TITLE
Modal standardization

### PR DESCRIPTION
**Pull request by b2simms**
Standardizing the modal template. Version two: getting one modal to display properly (not duplicate) with multiple renders on a single screen. Test as admin: http://127.0.0.1:3000/en/assignments/2/annotation_categories

### CODE REVIEW

**Comment by b2simms** [comment]
@david-yz-liu Just pushed the code that I used for styling the download modal for Annotation_Categories.   To test: http://127.0.0.1:3000/en/assignments/2/annotation_categories then click on 'download' in the top right corner. I implemented the modal using a common partial and reuse of the content_for tags in app/views/layouts/_modal_dialog.html. I will probably divide the code for the upload and download partials as you suggested, but it would be nice to reuse the variable names set in Annotation_Categories/index.html

**Comment by b2simms** [comment]
Commit e1f97d8: Third and fourth version of modals
Testing: http://127.0.0.1:3000/en/assignments/2/annotation_categories#
1.0 Click on 'Download' => should see the new modal
2.0 Click on 'Upload' => should see the upload modal with just the CSV content.
2.1 Click on 'Upload' then click the radio button for 'YML' => Should see YML content

**Comment by b2simms** [comment]
Commit 099177d: Modified Graders screen.
Testing: http://127.0.0.1:3000/en/assignments/2/graders#
1.0 Click on 'Download' => should see new modal

File: app/views/annotation_categories/index.html.erb [hunk 1]
```diff
@@ -40,79 +40,74 @@
...
+    <%= javascript_include_tag 'switch_content_control.js' %>
+    <%= javascript_tag do %>
+      LIB.init(["#annotation_upload_csv", "#annotation_upload_yml"]);
+    <% end %>
 
-  <%= t('annotations.upload.upload_message_yml').html_safe %>
-  <%= form_tag({ action: 'yml_upload',
-                 id: @assignment.id },
-               { multipart: true }) do %>
+    <p>CSV
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: Use `label_tag`  here

File: app/views/annotation_categories/index.html.erb [hunk 2]
```diff
@@ -40,79 +40,74 @@
...
-    <section class='dialog-actions'>
-      <%= submit_tag t(:upload),
-                     data: { disable_with: t(:uploading_please_wait) },
-                     id: 'upload_csv',
-                     disabled: true %>
-      <input type='reset' value='<%= t(:cancel) %>' onclick='modal_upload.close();' />
-    </section>
-  <% end %>
+    <!-- Script for switching content on radio buttons. -->
+    <%= javascript_include_tag 'switch_content_control.js' %>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: You can put all JS include tags at the top of the file.

File: app/views/annotation_categories/index.html.erb [hunk 3]
```diff
@@ -40,79 +40,74 @@
...
+                   { multipart: true }) do %>
+          <!-- Script for disabling upload buttons. -->
+          <%= javascript_include_tag 'upload_button_control.js',
+                                     id: 'upload_csv_script',
+                                     upload_id: 'annotation_category_list_csv',
+                                     button_id: 'upload_csv' %>
+          <p><%= file_field_tag :annotation_category_list_csv, size: 0 %></p>
+          <div>
+            <%= t('encoding') %>
+            <%= select_tag(:encoding, options_for_select(@encodings)) %>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: Please remove the parentheses here.

File: app/views/annotation_categories/index.html.erb [hunk 4]
```diff
@@ -40,79 +40,74 @@
   </div>
 </div>
 
+<%# Upload modal %>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: I think we talked about this before, but you should work on moving this whole piece into a new partial.

File: app/views/assignments/index.html.erb [hunk 5]
```diff
@@ -5,12 +5,12 @@
 <% content_for :title, t(:manage_course_work) %>
 
 <% @heading_buttons = [
-  { link_text: t('download'),
+  {  link_text: t('download'),
```
Comment (david-yz-liu) [inline app/views/assignments/index.html.erb]: No changes to this file should be made.

File: config/routes.rb [hunk 6]
```diff
@@ -232,8 +232,7 @@
           get 'add_grader_to_grouping'
           post 'csv_upload_grader_groups_mapping'
           post 'csv_upload_grader_criteria_mapping'
-          get 'download_grader_groupings_mapping'
-          get 'download_grader_criteria_mapping'
+          get 'download'
```
Comment (david-yz-liu) [inline config/routes.rb]: Run the tests locally yourself. You'll need to update a bunch to use the new route.

File: app/views/layouts/_modal_dialog_download.html.erb [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<%= stylesheet_link_tag('clickable') %>
```
Comment (david-yz-liu) [inline app/views/layouts/_modal_dialog_download.html.erb]: You can include this as part of the standard set of CSS includes for the application.

**Comment by b2simms** [comment]
Commit e4c8b46: Trying to resolve display issue. 
Currently working in annotation_categories. 
To test: http://127.0.0.1:3000/en/assignments/2/annotation_categories 
1. In views/annotation_categories/index.html: swap which modal is rendered first - line 45 and 48 respectively

File: config/routes.rb (hunk 6 above)
Comment (b2simms) [inline config/routes.rb]: Ah I see what you mean. I am refactoring this now

File: app/views/layouts/_modal_dialog.html.erb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<aside class='dialog' id=<%= content_for :modal_id %> >
```
Comment (david-yz-liu) [inline app/views/layouts/_modal_dialog.html.erb]: Should change this to `yield_content!` as well.

File: app/views/assignments/index.html.erb [hunk 9]
```diff
@@ -7,10 +7,10 @@
 <% @heading_buttons = [
   { link_text: t('download'),
     link_path: '#',
-    html_options: { onclick: 'modal_download.open()' } },
-  { link_text: t('upload'),
-    link_path: '#',
-    html_options: { onclick: 'modal_upload.open()' } }
+    html_options: { onclick: 'modal_download.open()' }},
+  {  link_text: t('upload'),
```
Comment (david-yz-liu) [inline app/views/assignments/index.html.erb]: Please undo these spacing changes.

File: app/views/annotation_categories/index.html.erb [hunk 10]
```diff
@@ -40,79 +41,10 @@
   </div>
 </div>
 
+<%# Download modal %>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: I don't think you need two separate comments... maybe just "Modals"?

File: app/views/annotation_categories/index.html.erb [hunk 11]
```diff
@@ -4,6 +4,7 @@
              handlers: [:erb] %>
   <%= javascript_include_tag 'https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' %>
   <%= javascript_include_tag 'MathJax/mathjax_helpers' %>
+  <%= stylesheet_link_tag('clickable')%>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/index.html.erb]: Please put a space after the close paren

File: app/views/annotation_categories/_annotation_upload_modal.html.erb [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+<%= content_for :modal_content do %>
+
+    <!-- Script for switching content on radio buttons. -->
+    <%= javascript_include_tag 'switch_content_control.js' %>
+    <%= javascript_tag do %>
+        LIB.init(["#annotation_upload_csv", "#annotation_upload_yml"]);
+    <% end %>
+
+    <%= label_tag :csv, "CSV"  %>
+      <%= radio_button_tag :upload_file_type, "csv", checked: true;  %>
```
Comment (david-yz-liu) [inline app/views/annotation_categories/_annotation_upload_modal.html.erb]: Please remove the semi-colon here

File: app/views/annotation_categories/_annotation_download_modal.html.erb [hunk 13]
```diff
@@ -0,0 +1,32 @@
+<%= content_for :modal_id, 'download_dialog' %>
+<%= content_for :modal_title, t('annotations.download.download_annotations') %>
+<%= content_for :modal_content do %>
+
+    <div class="clickable_links">
+      <div class="clickable_text">
+        <img style="width: 15px; height: 15px; margin-right: 15px;" src="<%= image_url("icons/download-file.png") %>">
```
Comment (david-yz-liu) [inline app/views/annotation_categories/_annotation_download_modal.html.erb]: Please make this a class.

File: app/views/layouts/_modal_dialog.html.erb [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+  <h2>
+    <%= yield_content! :modal_title %>
+  </h2>
+  <br/>
+  <%= raw yield_content! :modal_content %>
+</aside>
+<%# Flush content from variables. Allows reuse of partials %>
+<% content_for :modal_id, '', flush: true %>
```
Comment (david-yz-liu) [inline app/views/layouts/_modal_dialog.html.erb]: You should be able to get rid of these three lines.

File: app/views/layouts/_modal_dialog.html.erb [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<aside class='dialog' id=<%= content_for :modal_id %> >
+  <h2>
+    <%= yield_content! :modal_title %>
+  </h2>
+  <br/>
+  <%= raw yield_content! :modal_content %>
```
Comment (david-yz-liu) [inline app/views/layouts/_modal_dialog.html.erb]: @b2simms I just got rid of the `raw` and both dialogs seemed to work fine for me. Can you please try again after making the other changes?

**Pull request merged**